libweston/backend-rdp/rdprail.c

```diff
@@ -2959,136 +3228,145 @@ rdp_rail_peer_activate(freerdp_peer* client)
-	if (peerCtx->disp_server_context->DisplayControlCaps(peerCtx->disp_server_context) != CHANNEL_RC_OK)
+	if (disp_ctx->DisplayControlCaps(disp_ctx) != CHANNEL_RC_OK)
 		goto error_exit;
 
 	/* open HiDef (aka rdpgfx) channel. */
-	peerCtx->rail_grfx_server_context = rdpgfx_server_context_new(peerCtx->vcm);
-	if (!peerCtx->rail_grfx_server_context)
+	gfx_ctx = rdpgfx_server_context_new(peer_ctx->vcm);
+	if (!gfx_ctx)
 		goto error_exit;
-	peerCtx->rail_grfx_server_context->custom = (void*)client;
-	peerCtx->rail_grfx_server_context->CapsAdvertise = rail_grfx_client_caps_advertise;
-	peerCtx->rail_grfx_server_context->CacheImportOffer = rail_grfx_client_cache_import_offer;
-	peerCtx->rail_grfx_server_context->FrameAcknowledge = rail_grfx_client_frame_acknowledge;
-	if (!peerCtx->rail_grfx_server_context->Open(peerCtx->rail_grfx_server_context))
+	peer_ctx->rail_grfx_server_context = gfx_ctx;
+	gfx_ctx->custom = client;
+	gfx_ctx->CapsAdvertise = rail_grfx_client_caps_advertise;
+	gfx_ctx->CacheImportOffer = rail_grfx_client_cache_import_offer;
+	gfx_ctx->FrameAcknowledge = rail_grfx_client_frame_acknowledge;
+	if (!gfx_ctx->Open(gfx_ctx))
 		goto error_exit;
 	rail_grfx_server_opened = TRUE;
 
 #ifdef HAVE_FREERDP_GFXREDIR_H
+	GfxRedirServerContext *redir_ctx;
 	/* open Graphics Redirection channel. */
 	if (b->use_gfxredir) {
-		peerCtx->gfxredir_server_context = b->gfxredir_server_context_new(peerCtx->vcm);
-		if (!peerCtx->gfxredir_server_context)
+
+		redir_ctx = b->gfxredir_server_context_new(peer_ctx->vcm);
+		if (!redir_ctx)
 			goto error_exit;
-		peerCtx->gfxredir_server_context->custom = (void*)client;
-		peerCtx->gfxredir_server_context->GraphicsRedirectionLegacyCaps = gfxredir_client_graphics_redirection_legacy_caps;
-		peerCtx->gfxredir_server_context->GraphicsRedirectionCapsAdvertise = gfxredir_client_graphics_redirection_caps_advertise;
-		peerCtx->gfxredir_server_context->PresentBufferAck = gfxredir_client_present_buffer_ack;
-		if (peerCtx->gfxredir_server_context->Open(peerCtx->gfxredir_server_context) != CHANNEL_RC_OK)
+		peer_ctx->gfxredir_server_context = redir_ctx;
+		redir_ctx->custom = client;
+		redir_ctx->GraphicsRedirectionLegacyCaps = gfxredir_client_graphics_redirection_legacy_caps;
+		redir_ctx->GraphicsRedirectionCapsAdvertise = gfxredir_client_graphics_redirection_caps_advertise;
+		redir_ctx->PresentBufferAck = gfxredir_client_present_buffer_ack;
+		if (redir_ctx->Open(redir_ctx) != CHANNEL_RC_OK)
 			goto error_exit;
 		gfxredir_server_opened = TRUE;
 	}
-#endif // HAVE_FREERDP_GFXREDIR_H
+#endif /* HAVE_FREERDP_GFXREDIR_H */
 
 #ifdef HAVE_FREERDP_RDPAPPLIST_H
+	RdpAppListServerContext *applist_ctx;
 	/* open Application List channel. */
-	if (b->rdprail_shell_api &&
-		b->rdprail_shell_name &&
-		b->use_rdpapplist) {
-		peerCtx->applist_server_context = b->rdpapplist_server_context_new(peerCtx->vcm);
-		if (!peerCtx->applist_server_context)
+	if (api && b->rdprail_shell_name && b->use_rdpapplist) {

```

Comment:
Is "api" declared ?

Reply:
No, I missed that here - can I just elide the check for api though? nothing appears to use it in this block.

Reply:
Have removed it for now - will put it back if that wasn't a good idea.